staff/views.py

```diff
@@ -267,11 +277,131 @@ def make_announcement(request):
+            form_distribution.submitted = form.cleaned_data.get("submitted")
+            form_distribution.save()
+        return redirect("form_overview")
+    outstanding_form_dict = get_outstanding_forms()
+    form = CollectForms()
+    return render(request, "form_overview.html", {"outstanding_form_dict": outstanding_form_dict,
+                                                  "form": form})
+
+
+@group_required("staff")
+def notify_unsubmitted_users(request):
+    if request.method == "POST":
+        form = NotifyUnsubmittedUsersForm(request.POST)
+        if form.is_valid():
+            form_id = Form.objects.get(name=request.POST.get("notify_about")).id
+            form_distributions = FormDistribution.objects.filter(form_id=form_id, submitted=False)
+            for form_dist in form_distributions:
+                create_form_notification(request, form, form_dist.user_id)
+            if form.cleaned_data.get("email_recipients"):
+                email_list = get_emails_from_form_distributions(form_distributions)
+                email_form_notification(request, form, email_list)
+            messages.add_message(request, messages.SUCCESS, "Successfully Notified Users")
+            return redirect("staff")
+    form = NotifyUnsubmittedUsersForm()
+    notify_about = request.GET.get("notify_unsubmitted_users")
+    return render(request, "notify_unsubmitted_users.html", {"form": form,
+                                                             "notify_about": notify_about})
+
+
+@group_required("staff")
+def create_form_notification(request, form, user_id):
+    notification = Notification(
+        subject=form.cleaned_data.get("subject"),
+        notification=form.cleaned_data.get("notification"),
+        email_recipients=form.cleaned_data.get("email_recipients"),
+        created_by=DjangoUser.objects.get(id=request.user.id),
+        form_id=Form.objects.get(name=request.POST.get("notify_about")).id,
+        user_id=user_id
+    )
+    notification.save()
+
+
+@group_required("staff")
+def collect_forms(request):
+    return None
+
+
+@group_required("staff")
+def create_esign(request):
+    if request.method == "POST":
+        form = CreateEsignForm(request.POST)
+        if form.is_valid():
+            esign = Esign(
+                name=form.cleaned_data.get("name"),
+                template=form.cleaned_data.get("link"),
+                created_by=DjangoUser.objects.get(id=request.user.id)
+            )
+            esign.save()
+    form = CreateEsignForm(
+        initial={"created_by": DjangoUser.objects.get(id=request.user.id)}
+    )
+    return render(request, "create_esign.html", {"form": form})
+
+
 @group_required("staff")
 def my_account_staff(request):
     return render(request, "my_account_staff.html")
 
 
+@group_required("staff")
+def download_form_staff(request):
+    path = request.GET.get("path")
+    file_path = os.path.join(path)
```

Comment:
I'm not sure how to address this.  Will this depend on #22?

Reply:
The way to address it would be to sanitize the input first but #22 will make it irrelevant. We can’t really use local file system for storage under the assumption that there’s multiple web servers (the way most things get deployed these days).

Reply:
Ok I'll wait till #22 is finished to think about file storage again